geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/set/AbstractSScanIntegrationTest.java

```diff
@@ -323,26 +402,204 @@ public void givenNegativeCursor_returnsMembersUsingAbsoluteValueOfCursor() {
+  public void should_notErrorGivenNonzeroCursorOnFirstCall() {
+    List<String> set = initializeThreeMemberSet();
+
+    ScanResult<String> result = jedis.sscan(KEY, "5");
+
+    assertThat(result.getResult()).isSubsetOf(set);
+  }
+
+  @Test
+  public void should_notErrorGivenCountEqualToIntegerMaxValue() {
+    List<byte[]> set = initializeThreeMemberByteSet();
+
+    ScanParams scanParams = new ScanParams().count(Integer.MAX_VALUE);
+
+    ScanResult<byte[]> result =
+        jedis.sscan(KEY.getBytes(), ZERO_CURSOR.getBytes(), scanParams);
+    assertThat(result.getResult())
+        .containsExactlyInAnyOrderElementsOf(set);
+  }
+
+  @Test
+  public void should_notErrorGivenCountGreaterThanIntegerMaxValue() {
+    initializeThreeMemberByteSet();
+
+    String greaterThanInt = String.valueOf(2L * Integer.MAX_VALUE);
+    List<Object> result =
+        uncheckedCast(jedis.sendCommand(KEY.getBytes(), Protocol.Command.SSCAN,
+            KEY.getBytes(), ZERO_CURSOR.getBytes(),
+            "COUNT".getBytes(), greaterThanInt.getBytes()));
+
+    assertThat((byte[]) result.get(0)).isEqualTo(ZERO_CURSOR.getBytes());
+
+    List<byte[]> fields = uncheckedCast(result.get(1));
+    assertThat(fields).containsExactlyInAnyOrder(
+        FIELD_ONE.getBytes(),
+        FIELD_TWO.getBytes(),
+        FIELD_THREE.getBytes());
+  }
+
+  /**** Concurrency ***/
+
+  @Test
+  public void should_returnAllConsistentlyPresentMembers_givenConcurrentThreadsAddingAndRemovingMembers() {
+    final List<String> initialMemberData = makeSet();
+    jedis.sadd(KEY, initialMemberData.toArray(new String[initialMemberData.size()]));
+    final int iterationCount = 500;
+
+    Jedis jedis1 = jedis.getConnectionFromSlot(SLOT_FOR_KEY);
+    Jedis jedis2 = jedis.getConnectionFromSlot(SLOT_FOR_KEY);
+
+    new ConcurrentLoopingThreads(iterationCount,
+        (i) -> multipleSScanAndAssertOnContentOfResultSet(i, jedis1, initialMemberData),
+        (i) -> multipleSScanAndAssertOnContentOfResultSet(i, jedis2, initialMemberData),
+        (i) -> {
+          String field = "new_" + BASE_FIELD + i;
+          jedis.sadd(KEY, field);
+          jedis.srem(KEY, field);
+        }).run();
+
+    jedis1.close();
+    jedis2.close();
+  }
+
+  @Test
+  public void should_notAlterUnderlyingData_givenMultipleConcurrentSscans() {
+    final List<String> initialMemberData = makeSet();
+    jedis.sadd(KEY, initialMemberData.toArray(new String[initialMemberData.size()]));
+    final int iterationCount = 500;
+
+    Jedis jedis1 = jedis.getConnectionFromSlot(SLOT_FOR_KEY);
+    Jedis jedis2 = jedis.getConnectionFromSlot(SLOT_FOR_KEY);
+
+    new ConcurrentLoopingThreads(iterationCount,
+        (i) -> multipleSScanAndAssertOnContentOfResultSet(i, jedis1, initialMemberData),
+        (i) -> multipleSScanAndAssertOnContentOfResultSet(i, jedis2, initialMemberData))
+            .run();
+
+    initialMemberData
+        .forEach((member) -> assertThat(jedis.sismember(KEY, member)).isTrue());
```

Comment:
This would be better as:
```
assertThat(jedis.smembers(KEY)).containsExactlyInAnyOrderElementsOf(initialMemberData);
```
as we're currently only asserting that all the initial members are still present, but not that we haven't added any extra members somehow.